Clients/Xamarin.Interactive.Client/Telemetry/Client.cs

```diff
@@ -6,92 +6,120 @@
+                // We want to avoid having to perform a blocking Flush call on the UI thread, and since our events
+                // are currently few and far between, we set Capacity to 1 to essentially get auto-flush.
+                var channelBuffer = typeof (InMemoryChannel)
+                    .GetField ("buffer", BindingFlags.NonPublic | BindingFlags.Instance)
+                    .GetValue (channel);
+                channelBuffer
+                    .GetType ()
+                    .GetProperty ("Capacity", BindingFlags.Public | BindingFlags.Instance)
+                    .SetValue (channelBuffer, 1);
+
+                var config = new TelemetryConfiguration ("@TELEMETRY_INSTRUMENTATION_KEY@", channel);
+
+                appInsightsClient = new TelemetryClient (config);
+
+                appInsightsClient.Context.Session.Id = sessionid.ToString ();
+                appInsightsClient.Context.Device.OperatingSystem = host.OSName.ToString ();
+
+                // TODO: Make these GlobalProperties when we bump to 2.7.0-beta3 or later
+                var globalProperties = appInsightsClient.Context.Properties;
+                globalProperties.Add (
+                    "Product Version",
+                    BuildInfo.VersionString);
+                globalProperties.Add (
+                    "Build Hash",
+                    BuildInfo.Hash);
+                globalProperties.Add (
+                    "OS Platform",
+                    Runtime.CurrentProcessRuntime.OSPlatform.ToString ());
+                globalProperties.Add (
+                    "OS Architecture",
+                    RuntimeInformation.OSArchitecture.ToString ());
+                globalProperties.Add (
+                    "Process Architecture",
+                    Runtime.CurrentProcessRuntime.Architecture.ToString ());
+                globalProperties.Add (
+                    "Runtime Identifier",
+                    Runtime.CurrentProcessRuntime.RuntimeIdentifier);
+                globalProperties.Add (
+                    "OS Version",
+                    host.OSVersion.ToString ());
+                globalProperties.Add (
+                    "Release Candidate Level",
+                    ((byte)BuildInfo.Version.CandidateLevel).ToString ());
+                globalProperties.Add (
+                    "Release Candidate Level Name",
+                    BuildInfo.Version.CandidateLevel.ToString ().ToLowerInvariant ());
+                globalProperties.Add (
+                    "Machine ID",
+                    Sha256Hasher.Hash (MacAddressGetter.GetMacAddress ()));
+                globalProperties.Add (
+                    "Update Channel", updater.UpdateChannel);
             } catch (Exception e) {
-                Log.Error (TAG, e);
+                Log.Error (TAG, "Unable to create AppInsights client for telemetry", e);
```

Comment:
Any exception/error logging makes me nervous inside of the telem client because we could end up in a weird loop since exceptions will get logged through telemetry. I think we should have a `LogError` local function that always sets the SkipTelemetry flag.

Reply:
I can add one, sure. But also, I'm going to fix some `enabled` checks so it wouldn't matter.

Reply:
No real preference, I just don't want to end up in a weird state because we logged something through our normal means.